sql/mozfun/marketing/parse_campaign_name/udf.sql

```diff
@@ -0,0 +1,153 @@
+CREATE OR REPLACE FUNCTION marketing.parse_campaign_name(campaign_name STRING)
+RETURNS ARRAY<STRUCT<key STRING, value STRING>> AS (
+  CASE
+    WHEN REGEXP_CONTAINS(campaign_name, r"^(gads_v2)")
+      AND ARRAY_LENGTH(SPLIT(campaign_name, "_")) = 18
+      THEN mozfun.map.from_lists(
```

Comment:
Originally I had this living in `moz-fx-data-shared-prod.marketing_campaign_names` as separate UDFs (one per network/version), but because our staging workflow doesn't like it when I use shared-prod UDFs inside mozfun UDFs, I had to move them here. I've also deleted the UDFs at `marketing_campaign_names`(in fact, removed the entire dataset)